File: android_library/mobiledownloadlibrary/src/main/java/com/mediamonks/mobiledownload/model/ErrorModel.java
package com.mediamonks.mobiledownload.model;

/**
 * Created by raphael on 18/04/16.
 */
public class ErrorModel {

//        OTHER = 0,
//        NO_INTERNET_CONNECTION = 1,
//        NOT_ENOUGH_SPACE = 2,
//        CANCELED = 3,
//        PAUSED_NEED_CELLULAR_PERMISSION = 4,
//        PAUSED_NETWORK_UNAVAILABLE = 6

    public int mErrorType = 0;

    public String mErrorMessage;

    public DownloadItem[] mFiles;

    public ErrorModel() {
        // no-args constructor
    }
}

File: Assets/Plugins/MobileDownloadPlugin/Scripts/ErrorModel.cs
using UnityEngine;
using System.Collections;

public class ErrorModel {

	public enum ErrorType{
		OTHER = 0,
		NO_INTERNET_CONNECTION = 1,
		NOT_ENOUGH_SPACE = 2,
		CANCELED = 3,
		NETWORK_LOST = 5,
		TIMED_OUT = 7
	};

	public ErrorType mErrorType;

	public string mErrorMessage;

	public FileModel[] mFiles;

	public ErrorModel(){}

	public ErrorModel(string messageJson){
		JSONObject j = new JSONObject(messageJson);
		mErrorType = (ErrorType) j.GetField ("mErrorType").i;
		mErrorMessage = j.GetField ("mErrorMessage").ToString();
		if (j.GetField ("mFiles") != null && !j.GetField ("mFiles").ToString().Equals("null") && j.GetField ("mFiles").list != null) {
			mFiles = new FileModel[j.GetField ("mFiles").list.Count];
			for (int i = 0; i < j.GetField ("mFiles").list.Count; i++) {
				mFiles [i] = new FileModel (j.GetField ("mFiles").list [i].ToString ());
			}
		}
	}
}
